flink-clients/src/main/java/org/apache/flink/client/cli/ListOptions.java

```diff
@@ -28,20 +29,26 @@
  */
 public class ListOptions extends CommandLineOptions {
 
-	private final boolean running;
-	private final boolean scheduled;
+	private final boolean showRunning;
+	private final boolean showScheduled;
+	private final boolean showAll;
 
 	public ListOptions(CommandLine line) {
 		super(line);
-		this.running = line.hasOption(RUNNING_OPTION.getOpt());
-		this.scheduled = line.hasOption(SCHEDULED_OPTION.getOpt());
+		this.showAll = line.hasOption(ALL_OPTION.getOpt());
+		this.showRunning = line.hasOption(RUNNING_OPTION.getOpt()) || this.showAll;
+		this.showScheduled = line.hasOption(SCHEDULED_OPTION.getOpt()) || this.showAll;
```

Comment:
I would prefer this class being a dumb getter and having the OR logic in CLIFrontend.
This class is just a representation of what was configured, and I would leave the interpretation up to the using classes.